### PR TITLE
feat: update permissions

### PR DESCRIPTION
**Pull request by King-Mikaelson**
## What does this PR do?

This PR implements an endpoint for updating permissions for a specific role in an organization:

* Added PATCH /api/v1/organisations/:org_id/:role_id/permissions Endpoint: Updates permissions for a specific role in an organization.
* Authentication: Ensures only authenticated users can access this endpoint.
* Authorization: Implements OwnershipGuard to ensure only authorized users can update permissions.
* Validation: Implements input validation for the permission list obtained from the client.
* Error Handling: Returns specific HTTP status codes and messages based on various scenarios.

## Checklist:

* [x] Implemented `PATCH /api/v1/organisations/:org_id/:role_id/permissions` endpoint in `organisation-permissions.controller.ts`.
* [x] Implemented input validation using UpdatePermissionDto.
* [x] Tested endpoint functionality to ensure it correctly handles various scenarios.
* [x] Added appropriate HTTP status codes and response messages for edge cases.
* [x] Added Swagger documentation for the new endpoint.
* [x] Added unit tests for the `OrganisationPermissionsService`.
* [x] Updated API documentation to reflect the new endpoint.

## Response Example

**Success**

```json
{
  "message": "Permissions successfully updated",
  "status_code": 200
}
```

**Request Body**

```json
{
  "permission_list": {
    "canViewTransactions": false,
    "canViewRefunds": false,
    "canLogRefunds": false,
    "canViewUsers": false,
    "canCreateUsers": false,
    "canEditUsers": false,
    "canBlacklistWhitelistUsers": false
  }
}
```

## Link to Issue

Link to this issue: [[FEAT] Role-Based Access Control (RBAC) #[153]](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/153)

### CODE REVIEW

File: src/modules/organisation-permissions/organisation-permissions.service.spec.ts
Comment (Heba-WebDev) [inline src/modules/organisation-permissions/organisation-permissions.service.spec.ts:1]: Please move the mocks to the mocks folder, create one if it doesn't exist

**Review by Heba-WebDev — APPROVED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by Cobidex — DISMISSED** [review]

File: src/modules/organisation-permissions/dto/permission-list.dto.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { IsBoolean } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+import { PermissionCategory } from '../helpers/PermissionCategory';
+
+export class PermissionListDto {
+  [x: string]: any;
```
Comment (Cobidex) [inline src/modules/organisation-permissions/dto/permission-list.dto.ts:6]: Remove this

File: src/modules/organisation-permissions/helpers/custom-validator.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      }
+      if (typeof value !== 'boolean') {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  defaultMessage() {
```
Comment (Cobidex) [inline src/modules/organisation-permissions/helpers/custom-validator.ts:30]: Use a constant variable for this

Comment (King-Mikaelson) [reply src/modules/organisation-permissions/helpers/custom-validator.ts:30]: Hey bro I am not sure what I should do here, the default message is meant to be a function that is returned when validation fails.I can only be done in the function

File: src/modules/organisation-permissions/dto/permission-list.dto.ts (hunk 1 above)
Comment (King-Mikaelson) [reply src/modules/organisation-permissions/dto/permission-list.dto.ts]: Done

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by rugue — APPROVED** [review]

**Pull request merged**